### DISCUSSION TITLE
Using the baseUrl feature in 9.4 breaks my jest tests. Is there a way to get Jest to use the same config?

### DISCUSSION THREAD

**Discussion by atuttle** [vercel/next.js#12726] [Help] [answered]
I just read the announcement for next 9.4 (Congrats! 🎉 ) and decided to give it a shot.

I converted a bunch of `import '../../../components/Foo'` to `import 'components/Foo'`, and Next is working great. However, now my Jest tests fail because of the new pathing.

Is there some way to get Jest to use the same config?

**Comment by Timer** [comment, marked as answer]
As Next.js does not ship with Jest or a testing setup, you'll have to configure your custom Jest setup:
```json
"jest": {
    "moduleDirectories": ["node_modules", "src"],
   ...
}
```

**Reply by atuttle** [reply]
Where does this configuration go? jsconfig.json? jest.config.js? jest.setup.js? Elsewhere?

**Reply by Timer** [reply]
Where ever your existing config is! Please see the Jest documentation for more details:
https://jestjs.io/docs/en/configuration#moduledirectories-arraystring
https://jestjs.io/docs/en/configuration

**Reply by Timer** [reply]
@atuttle BTW! If you'd like to contribute, we'd be happy to accept a documentation and example update pointing users to Jest's `moduleDirectories` and `moduleNameMapper` when configuring their `jsconfig.json`.

I think it'd be best to cover this all in the [Jest example](https://github.com/zeit/next.js/tree/canary/examples/with-jest), then have our docs link to the example for reference.

**Reply by omarkhatibgg** [reply]
Hi @Timer  

I tried your snippet and still got `Cannot find module 'theme.config' from 'components/PreflightCSS.tsx'`

I tried both 
```js
"jest": {
    "moduleDirectories": ["node_modules", "."],
   ...
}
```
and 
```js
"jest": {
    "moduleDirectories": ["node_modules", "<rootDir>"],
   ...
}
```

Do you have a working example that could helps me ?

**Reply by Timer** [reply]
@omarkhatibco we'll need a full demo to help you identify the issue.

Our Jest example does not show how to use `moduleDirectories`, and we'd appreciate a PR once you get this working!

**Reply by omarkhatibgg** [reply]
@Timer  
you could check it here https://github.com/omarkhatibco/jest-with-typescript

I would love to make a small documentation on how to configure both Typescript and Jest!

**Reply by omarkhatibgg** [reply]
@Timer  also with `moduleNameMapper` it did not work.

**Reply by omarkhatibgg** [reply]
@Timer It's working now, there were an error naming my file `jest.config.ts` instead of `jest.config.js`

**Reply by omarkhatibgg** [reply]
Hi @Timer  using `"moduleDirectories": [".", "node_modules"],` will lead React to be undefined

I filed a bug report here #12922.

you can reproduce it here: https://github.com/omarkhatibco/jest-with-typescript

**Reply by omarkhatibgg** [reply]
I just fixed it today using this line of code: 

```js
jest: {
  moduleDirectories: ['node_modules', '<rootDir>'],
  }
```

**Comment by DrianHillman** [comment]
Just ran into this also and was able to fix this by updating my jest.config to include: 
```js
moduleDirectories: ['.', 'node_modules'],
```

Be sure to restart your server and re-run your `yarn/npm test` command if you're already in watch mode!

**Reply by DeBraid** [reply]
Tried above to no succes.

ended up following pattern similar to this: https://github.com/BenjaminWFox/nextjs-base/blob/master/jest.config.js

```
  moduleNameMapper: {
    '^@/components(.*)$': '<rootDir>/components$1'
  }
  ```
  
  Where   `moduleNameMapper` is added to `jest.config.js`.

**Comment by danilofuchs** [comment]
If you are using `ts-jest`, there is an included plugin which maps `paths` from tsconfig.json
https://kulshekhar.github.io/ts-jest/user/config/#jest-config-with-helper

**Reply by omarkhatibgg** [reply]
@danilofuchs Next already has a TS wrapper, you don't need `ts-jest`.

**Reply by lokenathsysalgo** [reply]
> Next already has a TS wrapper, you don't need ts-jest


@omarkhatibco I don't understand, how do I test typescript files without ts-jest. Please explain.
